module/os/linux/zfs/zvol_os.c

```diff
@@ -83,6 +117,108 @@ zv_request_task_free(zv_request_task_t *task)
 	kmem_free(task, sizeof (*task));
 }
 
+#ifdef HAVE_BLK_MQ
+/*
+ * This is our blk-mq workqueue callback function.  It's here that
+ * we process all the BIOs in a request.
+ */
+static void zvol_mq_work_func(struct work_struct *work)
+{
+	zv_work_t *zv_work;
+	struct bio *bio = NULL;
+	struct request *rq;
+	zvol_state_t *zv;
+	blk_status_t res = BLK_STS_OK;
+
+	zv_work = container_of(work, zv_work_t, work);
+	rq = zv_work->rq;
+	zv = rq->q->queuedata;
+
+	/* Tell the kernel that we are starting to process this request */
+	blk_mq_start_request(rq);
+
+	if (blk_rq_is_passthrough(rq)) {
+		/* Skip non filesystem request */
+		blk_mq_end_request(rq, BLK_STS_IOERR);
+		goto out;
+	}
+
+	/* Execute the BIOs in this request. */
+	__rq_for_each_bio(bio, rq) {
+		zvol_request_impl(zv, bio, 1);
+
+		/* Did this BIO cause an error?  If so, stop the request */
+#ifdef HAVE_BIO_BI_STATUS
+		res = bio->bi_status;
+#else
+		res = bio->bi_error;
+#endif
+		if (res != BLK_STS_OK) {
+			/* Got an error */
+			break;
```

Comment:
Shouldn't this be a `continue` so we process any other bios in the request?

Reply:
I don't think so - if one BIO is bad, then we need to mark the whole request as bad.